keras_nlp/src/models/video_swin/video_swin_layers.py

```diff
@@ -0,0 +1,1015 @@
+
+    Args:
+        hidden_dim (int): The number of units in the hidden layer.
+        output_dim (int): The number of units in the output layer.
+        drop_rate  (float): Float between 0 and 1. Fraction of the
+            input units to drop.
+        activation (str): Activation to use in the hidden layers.
+            Default is `"gelu"`.
+
+    References:
+        - [Video Swin Transformer](https://arxiv.org/abs/2106.13230)
+        - [Video Swin Transformer GitHub](https://github.com/SwinTransformer/Video-Swin-Transformer)
+    """
+
+    def __init__(
+        self, hidden_dim, output_dim, drop_rate=0.0, activation="gelu", **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.output_dim = output_dim
+        self.hidden_dim = hidden_dim
+        self._activation_identifier = activation
+        self.drop_rate = drop_rate
+        self.activation = layers.Activation(self._activation_identifier)
+        self.fc1 = layers.Dense(self.hidden_dim)
+        self.fc2 = layers.Dense(self.output_dim)
+        self.dropout = layers.Dropout(self.drop_rate)
+
+    def build(self, input_shape):
+        self.fc1.build(input_shape)
+        self.fc2.build((*input_shape[:-1], self.hidden_dim))
+        self.built = True
+
+    def call(self, x, training=None):
+        x = self.fc1(x)
+        x = self.activation(x)
+        x = self.dropout(x, training=training)
+        x = self.fc2(x)
+        x = self.dropout(x, training=training)
+        return x
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "output_dim": self.output_dim,
+                "hidden_dim": self.hidden_dim,
+                "drop_rate": self.drop_rate,
+                "activation": self._activation_identifier,
+            }
+        )
+        return config
+
+
+class VideoSwinPatchingAndEmbedding(keras.Model):
+    """Video to Patch Embedding layer for Video Swin Transformer models.
+
+    This layer performs the initial step in a Video Swin Transformer architecture by
+    partitioning the input video into 3D patches and embedding them into a vector
+    dimensional space.
+
+    Args:
+        patch_size (int): Size of the patch along each dimension
+            (depth, height, width). Default: (2,4,4).
+        embed_dim (int): Number of linear projection output channels. Default: 96.
+        norm_layer (keras.layers, optional): Normalization layer. Default: None
+
+    References:
+        - [Video Swin Transformer](https://arxiv.org/abs/2106.13230)
+        - [Video Swin Transformer GitHub](https://github.com/SwinTransformer/Video-Swin-Transformer)
+    """
+
+    def __init__(
+        self, patch_size=(2, 4, 4), embed_dim=96, norm_layer=None, **kwargs
+    ):
+        super().__init__(**kwargs)
+        self.patch_size = patch_size
+        self.embed_dim = embed_dim
+        self.norm_layer = norm_layer
+
+    def __compute_padding(self, dim, patch_size):
```

Comment:
Why no one underscore? Double underscore is usually reserved for python builtins